File: BOJ/Graph/Silver/2606/2606.py
import sys
from collections import deque
def bfs() :
    cnt = 0
    q = deque()
    q.append(1)
    visit[1] = True
    while len(q) > 0 :
        cur = q.popleft()
        for next in nodes[cur] :
            if visit[next] : continue
            visit[next] = True
            q.append(next)
            cnt+=1
    print(cnt)
n = int(input())
m = int(input())
nodes = [[] for _ in range(n+1)]
visit = [False for _ in range(n+1)]
for _ in range(m) :
    _to,_from = map(int,sys.stdin.readline().split())
    nodes[_to].append(_from)
    nodes[_from].append(_to)
bfs()

File: BOJ/Graph/Silver/2606/2606.swift
func bfs() {
    var cnt = 0
    var q : [Int] = []
    q.append(1)
    visit[1] = true
    while !q.isEmpty {
        let cur = q.removeFirst()
        for next in nodes[cur] {
            if visit[next] {continue}
            visit[next] = true
            q.append(next)
            cnt+=1
        }
    }
    print(cnt)
    }
let n = Int(readLine()!)!
let m = Int(readLine()!)!
var nodes : [[Int]] = Array(repeating: [], count: n+1)
var visit = Array(repeating: false, count: n+1)
for _ in 0..<m {
    let input = readLine()!.split(separator: " ").map{Int(String($0))!}
    nodes[input[0]].append(input[1])
    nodes[input[1]].append(input[0])
}
bfs()

